src/main/scala/scalismo/common/interpolation/LinearInterpolation.scala

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalismo.common.interpolation
+
+import scalismo.common.{ DiscreteField, Field }
+import scalismo.geometry.{ Dim, IntVector, IntVector2D, IntVector3D, NDSpace, Point, _1D, _2D, _3D }
+import scalismo.image.DiscreteImageDomain
+import scalismo.numerics.ValueInterpolator
+
+trait LinearImageInterpolator[D <: Dim, A] extends FieldInterpolator[D, DiscreteImageDomain[D], A] {
+
+  implicit def ndSpace: NDSpace[D]
```

Comment:
the two implicits should be made protected and in the concrete subclasses declared as private.